tests/Template/KeyOfTemplateTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psalm\Tests;
+
+use Psalm\Tests\Traits\InvalidCodeAnalysisTestTrait;
+use Psalm\Tests\Traits\ValidCodeAnalysisTestTrait;
+
+class KeyOfTemplateTest extends TestCase
+{
+    use InvalidCodeAnalysisTestTrait;
+    use ValidCodeAnalysisTestTrait;
+
+    /**
+     * @return iterable<string,array{code:string,assertions?:array<string,string>,ignored_issues?:list<string>,php_version?:string}>
+     */
+    public function providerValidCodeParse(): iterable
+    {
+        return [
+            'acceptsArrayKeysFn' => [
+                'code' => '<?php
+                    /**
+                     * @template T of array
+                     * @param T $array
+                     * @return key-of<T>[]
```

Comment:
seems like putting `value-of<T>[]` here breaks. I'd not have expected that

Reply:
There is no `TTemplateValueOf` currently.

Reply:
oh, right. Would that be easy to add? It feels like TValueOfArray is kinda meaningless on standard arrays without the possibility to combine this with templates

Reply:
Should be easy, I'll give it a try!

Reply:
I'll add this next week.